### PR TITLE
Close connection if stream id was not created

### PR DESCRIPTION
**Pull request by darkdarkdragon**
Right now, if `makeStreamID` returns empty string, connection just left hanging, so any any client (FFmpeg, OBS) just hangs trying to send data and not getting any response.

### CODE REVIEW

File: vidlistener/listener.go [hunk 1]
```diff
@@ -37,6 +37,8 @@ func (self *VidListener) HandleRTMPPublish(
 
 			strmID := makeStreamID(conn.URL)
 			if strmID == "" {
+				glog.Errorf("Error RTMP makeStreamID handler - empty stream ID")
```
Comment (j0sh) [inline vidlistener/listener.go]: The error should be logged by the `makeStreamID` caller so I don't think we need this here

Comment (darkdarkdragon) [reply vidlistener/listener.go]: Removed

**Comment by darkdarkdragon** [comment]
Added test

**Comment by j0sh** [comment]
LGTM. Squashed a fixup commit locally and rebased onto master.

**Pull request merged**